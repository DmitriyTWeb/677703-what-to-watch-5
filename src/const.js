export const RatingDesc = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const DEFAULT_RENDERED_FILMS_COUNT = 8;
export const PLAYING_TIMEOUT = 1000;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  MYLIST: `/mylist`,
  FILMS_ID: `/films/:id`,
  PLAYER_ID: `/player/:id`,
  REVIEW_ID: `/films/:id/review`,
  ERROR: `/error`,
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  PROMO: `/films/promo`,
};
