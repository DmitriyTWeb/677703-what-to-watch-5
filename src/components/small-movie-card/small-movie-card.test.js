import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card";
import {mockFilm} from "../../const.js";

jest.mock(`../video-player/video-player.jsx`, () => () => `VideoPlayer`);
jest.mock(`../../hocs/with-video-handling/with-video-handling`, () => () => `withVideoHandling`);

const noop = () => {};

it(`Should SmallMovieCard render correctly`, () => {
  const tree = renderer.create(
      <SmallMovieCard
        onMouseOver={noop}
        onMovieCardClick={noop}
        startTimer={noop}
        resetTimer={noop}
        onVideoMount={noop}
        film={mockFilm}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
