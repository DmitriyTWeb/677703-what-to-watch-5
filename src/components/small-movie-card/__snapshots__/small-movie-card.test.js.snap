// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SmallMovieCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <withVideoHandling
      coverImg="img/the-grand-budapest-hotel.jpg"
      onVideoMount={[Function]}
      src="https://some-link"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
